### PR TITLE
Feature 17 get boyler types by

### PR DESCRIPTION
**Pull request by pabloise**

### CODE REVIEW

File: src/controllers/boylerTypesController.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 const boylerTypes = require('../data/boylerTypes.json');
+const boylerTypesByDescription = require('../data/boylerTypes.json');
```
Comment (lmalvasia) [inline src/controllers/boylerTypesController.js:2]: This line it's not necessary. You have to use the const boylerTypes that you defined above.

File: src/controllers/boylerTypesController.js [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 
 const getAllBoylerTypes = () => boylerTypes;
+const getBoylerTypesByDescription = () => boylerTypesByDescription;
 
 module.exports = {
     getAllBoylerTypes
+};
+
+module.exports = {
```
Comment (lmalvasia) [inline src/controllers/boylerTypesController.js:11]: You should have one module.exports per file. If you have to export more than one thing, you will have something like:
```
module.exports = {
    getAllBoylerTypes,
   getBoylerTypesByDescription
};
```

File: src/controllers/boylerTypesController.js [hunk 3]
```diff
@@ -1,7 +1,13 @@
 const boylerTypes = require('../data/boylerTypes.json');
+const boylerTypesByDescription = require('../data/boylerTypes.json');
 
 const getAllBoylerTypes = () => boylerTypes;
+const getBoylerTypesByDescription = () => boylerTypesByDescription;
```
Comment (lmalvasia) [inline src/controllers/boylerTypesController.js:5]: You have to make a filter based on the boylerTypes const defined above. For example:
```
const getBoylerTypesByDescription = description => {
  const boylerTypesByDescription = boylerTypes.filter(boylerType => {
     // logic for filtering boylerTypes by description
  });
  return boylerTypesByDescription;
};
```

File: src/index.js [hunk 4]
```diff
@@ -15,6 +15,11 @@ app.get("/getAllBoylerTypes", (req, res) => {
   res.json(boylerTypes)
 });
 
+app.get("/getAllBoylerTypesByDescription/:description", (req, res) => {
+  const boylerTyperByDescription = boylerTypesController.getBoylerTypesByDescription();
```
Comment (lmalvasia) [inline src/index.js:19]: You have to pass the req.params.description to the getBoylerTypesByDescription function
```const boylerTyperByDescription = boylerTypesController.getBoylerTypesByDescription(description);```

File: src/index.js [hunk 5]
```diff
@@ -15,6 +15,11 @@ app.get("/getAllBoylerTypes", (req, res) => {
...
+  res.send(req.params.description)
```
Comment (lmalvasia) [inline src/index.js:20]: You should send the response from the `getBoylerTypesByDescription` function. 
```res.send(boylerTyperByDescription)```

**Review by lmalvasia — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (pabloise) [reply src/index.js]: Done

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**